modules/parent-join/src/main/java/org/elasticsearch/join/query/HasParentQueryBuilder.java

```diff
@@ -285,6 +285,10 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws I
     @Override
     protected void extractInnerHitBuilders(Map<String, InnerHitContextBuilder> innerHits) {
         if (innerHitBuilder != null) {
+            if (innerHits.containsKey(innerHitBuilder.getName())) {
```

Comment:
See above, we should check with the inferred name as well if innerHitBuilder.getName is null.